panic: runtime error: invalid memory address or nil pointer dereference in ignore.go

Looks like we're trying to do matches on a nil `*ignore.Matcher` for some reason.

```
[5FPON] 09:38:34 INFO: syncthing v0.14.54 "Dysprosium Dragonfly" (go1.11 windows-amd64) teamcity@build.syncthing.net 2018-12-05 11:18:45 UTC
...
[5FPON] 09:38:36 WARNING: Error on folder "foo" (hhsyf-pzy4t): folder marker missing
[5FPON] 09:38:36 INFO: Failed initial scan of receiveonly folder "foo" (hhsyf-pzy4t)
[5FPON] 09:38:36 INFO: GUI and API listening on 127.0.0.1:8384
[5FPON] 15:19:29 INFO: Puller (folder "bar" (default), file "AppData\\Local\\Temp\\baz_20180927154235.log"): finisher: pull: peers who had this file went away, or the file has changed while syncing. will retry later
Panic at 2018-12-21T15:19:33+01:00
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x80 pc=0x92a8f2]

goroutine 60 [running]:
github.com/syncthing/syncthing/lib/ignore.(*Matcher).Match(0x0, 0xc017ace900, 0x8d, 0xc017ace900)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/ignore/ignore.go:196 +0x62
github.com/syncthing/syncthing/lib/model.(*sendReceiveFolder).deleteDir(0xc0000cf080, 0xc01bef2c40, 0x70, 0x0, 0xc00890ede0, 0x0, 0x0)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/model/folder_sendrecv.go:1815 +0x5df
github.com/syncthing/syncthing/lib/model.(*sendReceiveFolder).performFinish(0xc0000cf080, 0x0, 0xc01bef2c40, 0x70, 0x0, 0x0, 0x1ed, 0x5b7acb34, 0x2ffb4d0, 0x6080a29ac0b9a6a6, ...)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/model/folder_sendrecv.go:1489 +0x4c8
github.com/syncthing/syncthing/lib/model.(*sendReceiveFolder).renameFile(0xc0000cf080, 0xc024e5f770, 0x44, 0x0, 0x0, 0x1a4, 0x53184d74, 0x9d5d730, 0xed4297a372b3846a, 0x0, ...)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/model/folder_sendrecv.go:915 +0x962
github.com/syncthing/syncthing/lib/model.(*sendReceiveFolder).processNeeded(0xc0000cf080, 0xc00008e750, 0xc000256780, 0xc00890ef60, 0xc00890eea0, 0xc00890ef00, 0xc00890ede0, 0xe16b00, 0xc00510c210, 0x60, ...)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/model/folder_sendrecv.go:476 +0xbf7
github.com/syncthing/syncthing/lib/model.(*sendReceiveFolder).pullerIteration(0xc0000cf080, 0xc00008e750, 0xc000256780, 0xc0000cf001, 0xc00890ede0, 0x2)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/model/folder_sendrecv.go:271 +0x58d
github.com/syncthing/syncthing/lib/model.(*sendReceiveFolder).pull(0xc0000cf080, 0xc000050a00)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/model/folder_sendrecv.go:185 +0x5f3
github.com/syncthing/syncthing/lib/model.(*folder).Serve(0xc0000cf080)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/lib/model/folder.go:136 +0xab0
github.com/syncthing/syncthing/vendor/github.com/thejerf/suture.(*Supervisor).runService.func1(0xc0001f5590, 0xc000000007, 0x14de0e8, 0xc0000cf080)
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/vendor/github.com/thejerf/suture/supervisor.go:540 +0x53
created by github.com/syncthing/syncthing/vendor/github.com/thejerf/suture.(*Supervisor).runService
	C:/BuildAgent/work/src/github.com/syncthing/syncthing/vendor/github.com/thejerf/suture/supervisor.go:528 +0x62
```

--- a/lib/model/folder_sendrecv.go
+++ b/lib/model/folder_sendrecv.go
@@ -473,7 +473,7 @@ nextFile:
 				// desired state with the delete bit set is in the deletion
 				// map.
 				desired := fileDeletions[candidate.Name]
-				if err := f.renameFile(candidate, desired, fi, dbUpdateChan, scanChan); err != nil {
+				if err := f.renameFile(candidate, desired, fi, ignores, dbUpdateChan, scanChan); err != nil {
 					// Failed to rename, try to handle files as separate
 					// deletions and updates.
 					break
@@ -815,7 +815,7 @@ func (f *sendReceiveFolder) deleteFile(file protocol.FileInfo, scanChan chan<- s
 
 // renameFile attempts to rename an existing file to a destination
 // and set the right attributes on it.
-func (f *sendReceiveFolder) renameFile(cur, source, target protocol.FileInfo, dbUpdateChan chan<- dbUpdateJob, scanChan chan<- string) error {
+func (f *sendReceiveFolder) renameFile(cur, source, target protocol.FileInfo, ignores *ignore.Matcher, dbUpdateChan chan<- dbUpdateJob, scanChan chan<- string) error {
 	// Used in the defer closure below, updated by the function body. Take
 	// care not declare another err.
 	var err error
@@ -912,7 +912,7 @@ func (f *sendReceiveFolder) renameFile(cur, source, target protocol.FileInfo, db
 	// of the source and the creation of the target temp file. Fix-up the metadata,
 	// update the local index of the target file and rename from temp to real name.
 
-	if err = f.performFinish(nil, target, curTarget, true, tempName, dbUpdateChan, scanChan); err != nil {
+	if err = f.performFinish(ignores, target, curTarget, true, tempName, dbUpdateChan, scanChan); err != nil {
 		return err
 	}
 

